crates/primitives/src/utilities.rs

```diff
@@ -31,6 +33,56 @@ pub fn create2_address(caller: B160, code_hash: B256, salt: U256) -> B160 {
     B160(hasher.finalize().as_slice()[12..].try_into().unwrap())
 }
 
+/// Calculates the `excess_blob_gas` from the parent header's `blob_gas_used` and `excess_blob_gas`.
+///
+/// See also [the EIP-4844 helpers](https://eips.ethereum.org/EIPS/eip-4844#helpers).
+#[inline]
+pub fn calc_excess_blob_gas(parent_excess_blob_gas: u64, parent_blob_gas_used: u64) -> u64 {
+    (parent_excess_blob_gas + parent_blob_gas_used).saturating_sub(TARGET_BLOB_GAS_PER_BLOCK)
```

Comment:
TODO for me, why `saturating_sub` this can go in negative. If we return zero does that mean that `base_blob_fee` is going down? Check `def calc_excess_blob_gas(parent: Header) -> int:` later in https://eips.ethereum.org/EIPS/eip-4844

Reply:
The `calc_excess_blob_gas` method defined there:
```python
def calc_excess_blob_gas(parent: Header) -> int:
    if parent.excess_blob_gas + parent.blob_gas_used < TARGET_BLOB_GAS_PER_BLOCK:
        return 0
    else:
        return parent.excess_blob_gas + parent.blob_gas_used - TARGET_BLOB_GAS_PER_BLOCK
```

Is the same thing as `saturating_sub`, because the first condition ensures that the function saturates at numeric bounds instead of returning any negative values.

Reply:
I see, thank you!